### PR TITLE
Enhance ui

### PR DESCRIPTION
**Pull request by ball6847**
Lots of improvement here, let me know if you don't want specific commits from this PR.

## Remove Broken Markup

- Remove `container` from stargazers component because it causes nested container, since it already wrap in `app.component.html`
- Remove `navbar-toggleable-md` , unused and causes unnecessary padding.

## Stargazer Improvements

- Add animation on new stargazer page.
- Add more grid for mobile device. (will show 2 items per row on small screen)
- Change items per page to 24 (originally 30) make it fit with grid (4 items per row on desktop and 2 items per row on small screen)
- Align content in center on mobile device. (follow the same direction as navbar-brand which also centered on small screen)

## *Side Effects From Above Changes

- `getStargazers()` method of `GithubService` now accept additional parameter `per_page`
- `StargazersComponent` has been refactored to reflect this change.
- Stargazer's specific style has been moved to the component itself.
- `GithubService`, `RepoDetail` now moved to `shared` folder.
- New shared animation also added to `shared` folder and used in `StargazersComponent`

**PS** - All tests are passed.
**PS2** - Consider using base href="/" for development and use `/frontend-war-angular` for github pages. So we can benefit from livereload from `ng serve` (which currently broken while  working on stargazers page)
**PS3** - From PS2 Consider using `useHash: true` on github pages, since any page reload in stargazer page will cause blank screen (Github pages not fully supported SPA yet, requests will not routed to index.html)

I can provide a PR for PS2 and PS3, just let me know if you want it.

Hope this help,
Porawit Poboonma

### CODE REVIEW

File: src/app/stargazers/stargazers.component.ts [hunk 1]
```diff
@@ -12,16 +15,16 @@ export class StargazersComponent implements OnInit {
   loading: boolean;
   private owner: string;
   private repo: string;
-  private page: number;
+  private page = 1;
+  private per_page = 24;
```
Comment (zixsma) [inline src/app/stargazers/stargazers.component.ts]: @ball6847 Could you change this variable to camelCase. Thank you.

**Comment by ball6847** [comment]
Done ;)

**Comment by zixsma** [comment]
@ball6847 Is there any way we can change base href by build prod or by CI? Any idea?

**Comment by ball6847** [comment]
We can use `--baseHref` option.

```sh
ng build --prod --baseHref /frontend-war-angular/
```

This should work.

**Comment by zixsma** [comment]
@ball6847 Could you make a change in travis config file for me. I think this solution is perfect because I've already added a script to copy index.html to 404.html after its build. Thank you very much.

**Comment by zixsma** [comment]
Oh and change default base href to `/` as you said too.

**Comment by ball6847** [comment]
Done, it should ready for both development and gh-pages now.

**Review by zixsma — APPROVED** [review]
Awesome

**Comment by zixsma** [comment]
@ball6847 Everything works flawlessly 👏 Thank you.

**Comment by ball6847** [comment]
@zixsma Thanks to you, now I know how to correctly deploy SPA to github pages.

**Pull request merged**